interface/src/app/gene/network/network.js

```diff
@@ -0,0 +1,262 @@
+/**
+ * "adage.gene.network" module.
+ */
+
+angular.module('adage.gene.network', [
+  'ui.router',
+  'placeholders',
+  'ui.bootstrap',
+  'ui.slider'
+])
+
+.config(function config($stateProvider) {
+  $stateProvider.state('gene_network', {
+    url: '/gene_network/?genes&mlmodel',
+    views: {
+      main: {
+        templateUrl: 'gene/network/network.tpl.html',
+        controller: 'GeneNetworkCtrl',
+        controllerAs: 'ctrl'
+      }
+    },
+    data: {pageTitle: 'Gene Network'}
+  });
+})
+
+.factory('EdgeService', ['$resource', function($resource) {
+  return $resource('/api/v0/edge/');
+}])
+
+.factory('NodeService', ['$resource', function($resource) {
+  return $resource('/api/v0/node/');
+}])
+
+.controller('GeneNetworkCtrl',
+   ['$stateParams', 'EdgeService', 'NodeService', '$log',
+    function GeneNetworkController($stateParams, EdgeService, NodeService,
+                                   $log) {
+      var rawMinWeight = -1.0;
+      var rawMaxWeight = 1.0;
+      var scaledMaxWeight = rawMaxWeight - rawMinWeight;
+      var self = this;
+      // The following properties of "self" will be available to HTMl.
+      self.minCorrelation = -1.0;
+      self.maxNodeNum = Number.MAX_SAFE_INTEGER;
+      self.statusMessage = 'Connecting to the server ...';
+      self.sliderOptions = {  // Slider configuration.
+        range: 'max',
```

Comment:
This **range** option in **ui-slider** controls which section is referred to by **.ui-slider-range** in network.less. It is subtle to explain but I will try. As you can see, **network.less* includes these lines:
```
 .ui-slider-range {
    background: DarkGray;
  }
```
When **range** is 'max', the color from the current slider to the ending point will have a background of DarkGray. On the other hand, if **range is 'min', then the color from the starting point to the current slider position will be DarkGray. See the page on my local web server:
http://165.123.67.202/#/gene_network/?genes=6546
and move the slider back and forth, see whether my explanation makes sense.